plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/InboundEndpointItemProvider.java

```diff
@@ -165,24 +173,33 @@ public InboundEndpointItemProvider(AdapterFactory adapterFactory) {
 
             switch (inboundEndpoint.getTransportJMSBrokerType()) {
             case WSO2_BROKER_PROFILE:
-                updateJavaNamingFactoryInitialProperty(inboundEndpoint, JAVA_NAMING_FACTORY_INITIAL_WSO2_BROKER);
-                updateJavaNamingProviderUrlProperty(inboundEndpoint, JAVA_NAMING_PROVIDER_URL_WSO2_BROKER);
-                updateTransportJMSDestinationProperty(inboundEndpoint, TRANSPORT_JMS_DESTINATION_WSO2_BROKER);
-                updateTransportJMSConnectionFactoryJNDINameProperty(inboundEndpoint,
-                        CONNECTION_FACTORY_JNDI_NAME_WSO2_BROKER);
+                if (currentJMSProfileType != JMSBrokerType.WSO2_BROKER_PROFILE || currentJMSProfileType == null) {
+                    updateJavaNamingFactoryInitialProperty(inboundEndpoint, JAVA_NAMING_FACTORY_INITIAL_WSO2_BROKER);
+                    updateJavaNamingProviderUrlProperty(inboundEndpoint, JAVA_NAMING_PROVIDER_URL_WSO2_BROKER);
+                    updateTransportJMSDestinationProperty(inboundEndpoint, TRANSPORT_JMS_DESTINATION_WSO2_BROKER);
+                    updateTransportJMSConnectionFactoryJNDINameProperty(inboundEndpoint,
+                            CONNECTION_FACTORY_JNDI_NAME_WSO2_BROKER);
+                    currentJMSProfileType = JMSBrokerType.WSO2_BROKER_PROFILE;
+                }
                 break;
             case ACTIVE_MQ:
-                updateJavaNamingFactoryInitialProperty(inboundEndpoint, JAVA_NAMING_FACTORY_INITIAL_ACTIVEMQ);
-                updateJavaNamingProviderUrlProperty(inboundEndpoint, JAVA_NAMING_PROVIDER_URL_ACTIVEMQ);
-                updateTransportJMSDestinationProperty(inboundEndpoint, TRANSPORT_JMS_DESTINATION_ACTIVEMQ);
-                updateTransportJMSConnectionFactoryJNDINameProperty(inboundEndpoint,
-                        CONNECTION_FACTORY_JNDI_NAME_ACTIVEMQ);
+                if (currentJMSProfileType != JMSBrokerType.ACTIVE_MQ || currentJMSProfileType == null) {
```

Comment:
Do we need the null equality check in here?

Reply:
Yes, otherwise the values will not get updated the first time it loads the model.

Reply:
Isn't that already covered with `currentJMSProfileType != JMSBrokerType.ACTIVE_MQ` check?

Reply:
Yes, it's redundant. Resolving.